src/QsCompiler/Core/SymbolTable.fs

```diff
@@ -980,22 +981,47 @@ and NamespaceManager
+                            | TupleType ts when ts.Length > 1 && argumentRestriction -> (decl.Position, t.Range |> orDefault |> QsCompilerDiagnostic.Error (ErrorCode.InnerTupleInEntryPointArgument, [])) |> Seq.singleton
+                            | ArrayType bt when argumentRestriction && bt.Type |> IsArray -> (decl.Position, t.Range |> orDefault |> QsCompilerDiagnostic.Error (ErrorCode.ArrayOfArrayInEntryPointArgument, [])) |> Seq.singleton
                             | _ -> Seq.empty)
-                        let inErrs = signature.Argument.Items.Select(snd) |> Seq.collect (validateArgAndReturnTypes true)
+                        let inErrs = signature.Argument.Items.Select snd |> Seq.collect (validateArgAndReturnTypes true)
                         let outErrs = signature.ReturnType |> validateArgAndReturnTypes false 
                         let signatureErrs = inErrs.Concat outErrs
-                        let hasCharacteristics = signature.Characteristics.Characteristics |> function | EmptySet | InvalidSetExpr -> false | _ -> true
-                        match Namespaces.TryGetValue parent.Namespace with
-                        | false, _ -> ArgumentException "no namespace with the given name exists" |> raise
-                        | true, ns when not ((ns.SpecializationsDefinedInAllSources parent.Name).Any(fst >> (<>)QsBody) || hasCharacteristics) -> ()
-                        | _ -> errs.Add (decl.Position, signature.Characteristics.Range.ValueOr decl.Range |> QsCompilerDiagnostic.Error (ErrorCode.InvalidEntryPointSpecialization, []))
+
+                        // validate entry point argument names
+                        let asCommandLineArg (str : string) = str.ToLowerInvariant() |> Seq.filter((<>)'_') |> String.Concat
+                        let reservedCommandLineArgs = [CommandLineArguments.SimulatorOption] |> List.map asCommandLineArg
+                        let nameAndRange (sym : QsSymbol) = sym.Symbol |> function
+                            | Symbol name -> Some (asCommandLineArg name.Value, sym.Range)
+                            | _ -> None
+                        let simplifiedArgNames = signature.Argument.Items.Select fst |> Seq.choose nameAndRange |> Seq.toList
+                        let verifyArgument i (arg, range : QsNullable<_>) = 
+                            if i > 0 && simplifiedArgNames.[..i-1] |> Seq.map fst |> Seq.contains arg
+                            then errs.Add (decl.Position, range.ValueOr decl.Range |> QsCompilerDiagnostic.Error (ErrorCode.DuplicateEntryPointArgumentName, []))
+                            elif reservedCommandLineArgs.Contains arg
+                            then errs.Add (decl.Position, range.ValueOr decl.Range |> QsCompilerDiagnostic.Error (ErrorCode.ReservedEntryPointArgumentName, []))
```

Comment:
This should be a warning instead of an error.

Reply:
Right, for backward compatibility when we add a new one. Done.